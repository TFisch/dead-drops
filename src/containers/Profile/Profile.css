.board-display {
  background-color: darkorange;
  width: 50em; }

.container {
  display: flex; 
  height: 46em;
}

.profile-image {
  border: 1px solid black;
  border-radius: 50%; }

.user-profile {
  background-color: magenta;
  height: 100%;
  width: 40em; }

.welcome-text {
  font-family: 'PT Sans', Arial, Verdana;
  font-weight: 400; }

.profile-wrap {
  align-items: center;
  display: flex; }

.post-wrap {
  align-items: center;
  display: flex;
  justify-content: space-between; }
