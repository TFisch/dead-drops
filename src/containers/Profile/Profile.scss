@import '../../styles/mixins.scss';

.board-display {
  background-color: darkorange;
  width: 50em;
}

.container {
  background-color: aqua;
  display: flex;
}

.profile-image {
  border: 1px solid black;
  border-radius: 50%;
}

.user-profile {
  background-color: magenta;
  height: 40em;
  width: 40em;
}

.welcome-text {
  @include primary-font;
}

.profile-wrap {
  align-items: center;
  display: flex;
}

.post-wrap {
  align-items: center;
  display: flex;
  justify-content: space-between;
}