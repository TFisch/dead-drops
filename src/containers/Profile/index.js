import React, { Component } from 'react';
import { getToken, setUser } from '../../actions';
import { connect } from 'react-redux';
import { fetchDropBoard } from '../../api'
import './Profile.css'
import DropSubmitForm from '../DropSubmitForm';
import { DropList } from '../../components/DropList';
import ConfirmDrop from '../ConfirmDrop';

export class Profile extends Component {
  constructor(props) {
    super(props)
    this.state = {
      username: '',
      image: '',
      dropFormActive: false,
      dropListRetrieved: false,
      dropListHidden: false,
      pictureRetrieved: false,
      retrievedBoard: [],
      dropToConfirm: false
    }
  }

  componentDidUpdate() {
    if (!this.state.dropListRetrieved) {
      this.retrieveDropBoard();
    }
  }

  retrieveDropBoard = async () => {
    const token = this.props.token
    const retrievedBoard = await fetchDropBoard(token);
    await this.setState({ retrievedBoard, dropListRetrieved: true })
  }

  handlePost = () => {
    const dropFormActive = !this.state.dropFormActive;
    this.setState({ dropFormActive })
  }

  toggleSubmit = (boolean) => {
    this.setState({ dropToConfirm: boolean, dropListHidden: true })
  }

  toggleDisplay = () => {
    const dropListHidden = !this.state.dropListHidden;
    this.setState({ dropListHidden })
  }

  render() {
    const { username, image } = this.props.user;
    const {
      dropFormActive,
      retrievedBoard,
      dropListRetrieved,
      dropToConfirm,
      dropListHidden
    } = this.state;

    return (
      <div className="container">
        <div className='user-profile'>
          <div className="post-wrap">
            <div className="profile-wrap">
              <img src={image} className='profile-image' alt='user-avatar' />
              <h1 className="welcome-text">Welcome, {username}!</h1>
            </div>
            {!dropFormActive && <button className="post-button" onClick={this.handlePost}>POST A DEAD DROP</button>}
          </div>
          {dropFormActive && <DropSubmitForm toggleSubmit={this.toggleSubmit} />}
        </div>
        <div className="board-display">
          {dropListRetrieved && !dropListHidden && < DropList retrievedBoard={retrievedBoard} />}
          {dropToConfirm && dropListHidden && <ConfirmDrop toggleDisplay={this.toggleDisplay} />}
        </div>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  token: state.token,
  user: state.user,
  location: state.locationData
});

export const mapDispatchToProps = (dispatch) => ({
  getToken: (token) => dispatch(getToken(token)),
  setUser: (user) => dispatch(setUser(user)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Profile);
