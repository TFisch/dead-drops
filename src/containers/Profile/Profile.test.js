import React from 'react';
import { shallow } from 'enzyme';
import { Profile } from '../Profile/'

describe('Profile', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Profile />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});