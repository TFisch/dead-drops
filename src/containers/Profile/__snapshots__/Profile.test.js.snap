// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Profile />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="user-profile"
>
          <div
                    className="post-wrap"
          >
                    <div
                              className="profile-wrap"
                    >
                              <img
                                        alt="user-avatar"
                                        className="profile-image"
                                        src=""
                              />
                              <h1>
                                        Welcome, 
                                        
                                        !
                              </h1>
                    </div>
                    <button
                              className="post-button"
                              onClick={[Function]}
                    >
                              POST A DEAD DROP
                    </button>
          </div>
</div>,
        <div
          className="board-display"
>
          책
</div>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="post-wrap"
>
              <div
                            className="profile-wrap"
              >
                            <img
                                          alt="user-avatar"
                                          className="profile-image"
                                          src=""
                            />
                            <h1>
                                          Welcome, 
                                          
                                          !
                            </h1>
              </div>
              <button
                            className="post-button"
                            onClick={[Function]}
              >
                            POST A DEAD DROP
              </button>
</div>,
            false,
          ],
          "className": "user-profile",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="profile-wrap"
>
                  <img
                                    alt="user-avatar"
                                    className="profile-image"
                                    src=""
                  />
                  <h1>
                                    Welcome, 
                                    
                                    !
                  </h1>
</div>,
                <button
                  className="post-button"
                  onClick={[Function]}
>
                  POST A DEAD DROP
</button>,
              ],
              "className": "post-wrap",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <img
                      alt="user-avatar"
                      className="profile-image"
                      src=""
/>,
                    <h1>
                      Welcome, 
                      
                      !
</h1>,
                  ],
                  "className": "profile-wrap",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "user-avatar",
                      "className": "profile-image",
                      "src": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Welcome, ",
                        "",
                        "!",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Welcome, ",
                      "",
                      "!",
                    ],
                    "type": "h1",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "POST A DEAD DROP",
                  "className": "post-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "POST A DEAD DROP",
                "type": "button",
              },
            ],
            "type": "div",
          },
          false,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "책",
            false,
          ],
          "className": "board-display",
        },
        "ref": null,
        "rendered": Array [
          "책",
          false,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="user-profile"
>
            <div
                        className="post-wrap"
            >
                        <div
                                    className="profile-wrap"
                        >
                                    <img
                                                alt="user-avatar"
                                                className="profile-image"
                                                src=""
                                    />
                                    <h1>
                                                Welcome, 
                                                
                                                !
                                    </h1>
                        </div>
                        <button
                                    className="post-button"
                                    onClick={[Function]}
                        >
                                    POST A DEAD DROP
                        </button>
            </div>
</div>,
          <div
            className="board-display"
>
            책
</div>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="post-wrap"
>
                <div
                                className="profile-wrap"
                >
                                <img
                                                alt="user-avatar"
                                                className="profile-image"
                                                src=""
                                />
                                <h1>
                                                Welcome, 
                                                
                                                !
                                </h1>
                </div>
                <button
                                className="post-button"
                                onClick={[Function]}
                >
                                POST A DEAD DROP
                </button>
</div>,
              false,
            ],
            "className": "user-profile",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="profile-wrap"
>
                    <img
                                        alt="user-avatar"
                                        className="profile-image"
                                        src=""
                    />
                    <h1>
                                        Welcome, 
                                        
                                        !
                    </h1>
</div>,
                  <button
                    className="post-button"
                    onClick={[Function]}
>
                    POST A DEAD DROP
</button>,
                ],
                "className": "post-wrap",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <img
                        alt="user-avatar"
                        className="profile-image"
                        src=""
/>,
                      <h1>
                        Welcome, 
                        
                        !
</h1>,
                    ],
                    "className": "profile-wrap",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "alt": "user-avatar",
                        "className": "profile-image",
                        "src": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "Welcome, ",
                          "",
                          "!",
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        "Welcome, ",
                        "",
                        "!",
                      ],
                      "type": "h1",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "POST A DEAD DROP",
                    "className": "post-button",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "POST A DEAD DROP",
                  "type": "button",
                },
              ],
              "type": "div",
            },
            false,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "책",
              false,
            ],
            "className": "board-display",
          },
          "ref": null,
          "rendered": Array [
            "책",
            false,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
